$NetBSD: patch-configure.ac,v 1.17 2020/08/11 16:07:39 ryoon Exp $

* Add NetBSD/aarch64 support.
* Disable freetype2 and fontconfig from pkgsrc explicitly.
  to fix X11_TYPE=native build and internal pdfium's internal freetype build.
* Fix some static links.
* Include header files from gpgme correctly.
* Find pkgsrc/devel/apache-ant.

--- configure.ac.orig	2020-07-29 19:29:17.000000000 +0000
+++ configure.ac
@@ -4593,6 +4593,11 @@ linux-android*)
         RTL_ARCH=X86_64
         PLATFORMID=netbsd_x86_64
         ;;
+    aarch64)
+        CPUNAME=AARCH64
+        RTL_ARCH=AARCH64
+        PLATFORMID=netbsd_aarch64
+        ;;
     *)
         AC_MSG_ERROR([Unsupported host_cpu $host_cpu for host_os $host_os])
         ;;
@@ -4992,6 +4997,7 @@ fi
 AC_SUBST(ENABLE_CUPS)
 
 # fontconfig checks
+#test_fontconfig=no
 if test "$test_fontconfig" = "yes"; then
     PKG_CHECK_MODULES([FONTCONFIG], [fontconfig >= 2.4.1])
     SYSTEM_FONTCONFIG=TRUE
@@ -8287,7 +8293,7 @@ else
     if test "$COM" = "MSC"; then
         LIBJPEG_LIBS="${WORKDIR}/UnpackedTarball/libjpeg-turbo/.libs/libjpeg.lib"
     else
-        LIBJPEG_LIBS="-L${WORKDIR}/UnpackedTarball/libjpeg-turbo/.libs -ljpeg"
+        LIBJPEG_LIBS="${WORKDIR}/UnpackedTarball/libjpeg-turbo/.libs/libjpeg.a"
     fi
 
     case "$host_cpu" in
@@ -8599,6 +8605,7 @@ fi
 dnl ===================================================================
 dnl Check whether freetype is available
 dnl ===================================================================
+#test_freetype=no
 if test  "$test_freetype" = "yes"; then
     AC_MSG_CHECKING([whether freetype is available])
     # FreeType has 3 different kinds of versions
@@ -8616,9 +8623,9 @@ if test  "$test_freetype" = "yes"; then
 else
     FREETYPE_CFLAGS="${ISYSTEM}${WORKDIR}/UnpackedTarball/freetype/include"
     if test "x$ac_config_site_64bit_host" = xYES; then
-        FREETYPE_LIBS="-L${WORKDIR}/UnpackedTarball/freetype/instdir/lib64 -lfreetype"
+        FREETYPE_LIBS="${WORKDIR}/UnpackedTarball/freetype/instdir/lib64/libfreetype.a"
     else
-        FREETYPE_LIBS="-L${WORKDIR}/UnpackedTarball/freetype/instdir/lib -lfreetype"
+        FREETYPE_LIBS="${WORKDIR}/UnpackedTarball/freetype/instdir/lib/libfreetype.a"
     fi
 fi
 AC_SUBST(FREETYPE_CFLAGS)
@@ -9884,11 +9891,11 @@ dnl HarfBuzz
 dnl ===================================================================
 libo_CHECK_SYSTEM_MODULE([graphite],[GRAPHITE],[graphite2 >= 0.9.3],
                          ["-I${WORKDIR}/UnpackedTarball/graphite/include -DGRAPHITE2_STATIC"],
-                         ["-L${WORKDIR}/LinkTarget/StaticLibrary -lgraphite"])
+                         ["${WORKDIR}/LinkTarget/StaticLibrary/libgraphite.a"])
 
 libo_CHECK_SYSTEM_MODULE([harfbuzz],[HARFBUZZ],[harfbuzz-icu >= 0.9.42],
                          ["-I${WORKDIR}/UnpackedTarball/harfbuzz/src"],
-                         ["-L${WORKDIR}/UnpackedTarball/harfbuzz/src/.libs -lharfbuzz"])
+                         ["${WORKDIR}/UnpackedTarball/harfbuzz/src/.libs/libharfbuzz.a"])
 
 if test "$COM" = "MSC"; then # override the above
     GRAPHITE_LIBS="${WORKDIR}/LinkTarget/StaticLibrary/graphite.lib"
@@ -11387,7 +11394,7 @@ elif test "$_os" = "Linux" -o "$_os" = "
         SYSTEM_GPGMEPP=TRUE
 
         # C++ library doesn't come with fancy gpgmepp-config, check for headers the old-fashioned way
-        AC_CHECK_HEADER(gpgme++/gpgmepp_version.h, [ GPGMEPP_CFLAGS=-I/usr/include/gpgme++ ],
+        AC_CHECK_HEADER(gpgme++/gpgmepp_version.h, [ GPGMEPP_CFLAGS=-I@PREFIX@/include/gpgme++ ],
             [AC_MSG_ERROR([gpgmepp headers not found, install gpgmepp development package])], [])
         # progress_callback is the only func with plain C linkage
         # checking for it also filters out older, KDE-dependent libgpgmepp versions
@@ -12454,8 +12461,8 @@ EOF
                     if test -f /usr/share/ant-core/lib/ant.jar; then
                         ANT_LIB=/usr/share/ant-core/lib
                     else
-                        if test -f $ANT_HOME/lib/ant/ant.jar; then
-                            ANT_LIB="$ANT_HOME/lib/ant"
+                        if test -f $ANT_HOME/lib/java/ant/ant.jar; then
+                            ANT_LIB="$ANT_HOME/lib/java/ant"
                         else
                             if test -f /usr/share/lib/ant/ant.jar; then
                                 ANT_LIB=/usr/share/lib/ant
